promql/parser/ast.go

```diff
@@ -125,10 +125,16 @@ type MatrixSelector struct {
 
 // SubqueryExpr represents a subquery.
 type SubqueryExpr struct {
-	Expr   Expr
-	Range  time.Duration
-	Offset time.Duration
-	Step   time.Duration
+	Expr  Expr
+	Range time.Duration
+	// OriginalOffset is the actual offset that was set in the query.
+	// This never changes.
+	OriginalOffset time.Duration
+	// Offset is the offset used during the query execution.
+	// This can change because of the @ modifier.
```

Comment:
> Same for subqueries, it is commented for that node too. What exactly is the confusion here?

My point is that this makes it seem like this can only change because of @, so not also mentioning that it can change due to subqueries is confusing.